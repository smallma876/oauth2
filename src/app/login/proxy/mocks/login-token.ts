export const loginToken = {
  iss: 'https://accounts.google.com',
  nbf: 1661831589,
  aud: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  sub: '105035299156813441187',
  email: 'sergio.mallma876@gmail.com',
  email_verified: true,
  azp: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  name: 'Sergio Mallma Chavez',
  picture: 'https://lh3.googleusercontent.com/a-/AFdZucq8mJ7sluN0hMjUBWZSj0LDYWfoN4KvV-52JpyI0A=s96-c',
  given_name: 'Sergio',
  family_name: 'Mallma Chavez',
  iat: 1661831889,
  exp: 1661835489,
  jti: '677749ac9724e83aa7155eac897c2135dfcb979d',
};
