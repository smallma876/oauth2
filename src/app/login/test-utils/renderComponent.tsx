import { act, render, RenderResult } from '@testing-library/react';
import { ReactElement } from 'react';
import { GlobalAppProvider } from '../../../store/app-context';
import { AppState, DispatchApp } from '../../../store/reducer';
import { LoginState } from '../store/reducer';
import * as context from './../store/login-context';

interface RenderComponentInterface {
  Component: ReactElement;
  stateProps?: Partial<AppState>;
  extraState?: Partial<LoginState>;
  dispatch?: DispatchApp;
}

export const renderComponent = async (renderComponentConfig: RenderComponentInterface) => {
  let component = {} as RenderResult;
  const { stateProps, dispatch, Component, extraState = {} } = renderComponentConfig;

  await act(async () => {
    component = render(
      <GlobalAppProvider testStateProps={stateProps} testDispatch={dispatch}>
        <context.LoginProvider extraState={extraState}>{Component}</context.LoginProvider>
      </GlobalAppProvider>,
    );
  });

  return component;
};
