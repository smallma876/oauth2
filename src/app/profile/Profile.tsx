import React, { lazy, ReactNode, Suspense } from 'react';
import { Route, Routes } from 'react-router-dom';
import { useGlobalAppState } from '../../store/app-context';

const UserInformation = lazy(() => import('./pages/UserInformation/UserInformation'));

const Profile: React.FC = () => {
  const { user } = useGlobalAppState();
  const WithRedirect = ({ children }: { children: ReactNode }) => {
    if (!user) {
      window.location.href = '/b';
      return <></>;
    }
    return <>{children}</>;
  };

  return (
    <Suspense>
      <Routes>
        <Route
          path="/"
          element={
            <WithRedirect>
              <UserInformation />
            </WithRedirect>
          }
        />
      </Routes>
    </Suspense>
  );
};

export default Profile;
