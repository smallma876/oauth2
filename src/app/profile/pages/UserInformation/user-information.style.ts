import styled from 'styled-components';
import { getMediaQuery, MediaQuerys } from '../../../../shared/utils/media-querys';

const CardProfile = styled.div`
  ${getMediaQuery(MediaQuerys.largeTablet)} {
    padding: 10px;
    position: relative;
    border-radius: 5px;
    box-shadow: 0 35px 80px rgba(0, 0, 0, 0.15);
    max-width: 500px;
    min-width: 520px;
    min-height: 300px;
    background: #fff;
  }
`;

const ImageWrapper = styled.div`
  position: absolute;
  left: 50%;
  top: -29%;
  transform: translateX(-50%);
  width: 150px;
  height: 150px;
  background: #fff;
  border-radius: 20px;
  box-shadow: 0 15px 50px rgba(0, 0, 0, 0.35);
  overflow: hidden;

  & img {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;

const Image = styled.img``;

export const UserInformationStyled = {
  CardProfile,
  ImageWrapper,
  Image,
};
