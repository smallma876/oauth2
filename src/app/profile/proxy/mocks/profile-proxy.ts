export const mockAlbum = [
  {
    name: 'first',
    url: 'http',
  },
];
