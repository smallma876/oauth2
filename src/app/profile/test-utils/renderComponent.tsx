import { act, render, RenderResult } from '@testing-library/react';
import { ReactElement } from 'react';
import { GlobalAppProvider } from '../../../store/app-context';
import { AppState, DispatchApp } from '../../../store/reducer';
import * as context from '../store/profile-context';
import { ProfileState } from '../store/reducer';

interface RenderComponentInterface {
  Component: ReactElement;
  stateProps?: Partial<AppState>;
  extraState?: Partial<ProfileState>;
  dispatch?: DispatchApp;
}

export const renderComponent = async (renderComponentConfig: RenderComponentInterface) => {
  let component = {} as RenderResult;
  const { stateProps, dispatch, Component, extraState = {} } = renderComponentConfig;

  await act(async () => {
    component = render(
      <GlobalAppProvider testStateProps={stateProps} testDispatch={dispatch}>
        <context.ProfileProvider extraState={extraState}>{Component}</context.ProfileProvider>
      </GlobalAppProvider>,
    );
  });

  return component;
};
