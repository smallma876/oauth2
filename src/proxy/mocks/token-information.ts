export const TokenInformation = {
  aud: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  azp: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  email: 'sergio.mallma876@gmail.com',
  email_verified: true,
  exp: 1662353890,
  family_name: 'Mallma Chavez',
  given_name: 'Sergio',
  iat: 1662350290,
  iss: 'https://accounts.google.com',
  jti: 'ae3f781891bbad12d13c9d411ad5a45e48ccf556',
  name: 'Sergio Mallma Chavez',
  nbf: 1662349990,
  picture: 'https://lh3.googleusercontent.com/a-/AFdZucr57cAutaV8GJ70u1HWwolYmglPELDZSCrfGEFCVQ=s96-c',
  sub: '105035299156813441187',
};
