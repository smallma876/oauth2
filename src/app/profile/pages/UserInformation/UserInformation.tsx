import jwtDecode from 'jwt-decode';
import React, { useEffect, useState } from 'react';
import { useGlobalAppState } from '../../../../store/app-context';
import { LoginToken } from '../../../login/domain/login-token';
import { profileProxy } from '../../proxy/profile-proxy';
import { UserInformationStyled } from './user-information.style';

const UserInformation: React.FC = () => {
  const { user, token } = useGlobalAppState();

  useEffect(() => {
    const getAlbums = async () => {
      await profileProxy.getAlbums(token);
    };

    getAlbums();

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <UserInformationStyled.CardProfile>
      <UserInformationStyled.ImageWrapper>
        <UserInformationStyled.Image src={user!.picture} alt="profile" />
      </UserInformationStyled.ImageWrapper>
      <div>{user!.name}</div>
    </UserInformationStyled.CardProfile>
  );
};

export default UserInformation;
