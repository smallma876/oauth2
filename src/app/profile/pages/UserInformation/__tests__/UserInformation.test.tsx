import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { TokenInformation } from '../../../../../proxy/mocks/token-information';
import Profile from '../../../Profile';
import { renderComponent } from '../../../test-utils/renderComponent';
import UserInformation from '../UserInformation';

describe('UserInformation test', () => {
  it('render user information', async () => {
    const { container } = await renderComponent({
      stateProps: {
        user: TokenInformation,
      },
      Component: (
        <MemoryRouter initialEntries={['/']}>
          <Profile />
        </MemoryRouter>
      ),
    });
    expect(container).toMatchSnapshot();
  });

  it('should redirect to login', async () => {
    const { container } = await renderComponent({
      stateProps: {
        user: null,
      },
      Component: (
        <MemoryRouter initialEntries={['/']}>
          <Profile />
        </MemoryRouter>
      ),
    });
    expect(container).toMatchSnapshot();
  });
});
