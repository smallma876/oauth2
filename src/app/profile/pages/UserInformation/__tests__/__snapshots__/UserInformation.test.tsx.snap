// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInformation test render user information 1`] = `
<div>
  <div
    class="sc-bczRLJ dXtvxe"
  >
    <div
      class="sc-gsnTZi fPmQmu"
    >
      <img
        alt="profile"
        class="sc-dkzDqf"
        src="https://lh3.googleusercontent.com/a-/AFdZucr57cAutaV8GJ70u1HWwolYmglPELDZSCrfGEFCVQ=s96-c"
      />
    </div>
    <div>
      Sergio Mallma Chavez
    </div>
  </div>
</div>
`;

exports[`UserInformation test should redirect to login 1`] = `<div />`;
